GVFS/GVFS/CommandLine/UnmountVerb.cs

```diff
@@ -55,22 +54,28 @@ public override void Execute()
                 this.AcquireLock(root);
             }
 
-            if (!this.ShowStatusWhileRunning(
+            string enlistmentId;
+            string mountId;
+            GVFSVerb.ForExistingEnlistment.LookupIds(enlistment, out enlistmentId, out mountId);
+            using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "GVFSUnmount", enlistmentId, mountId))
+            {
+                if (!this.ShowStatusWhileRunning(
                 () => { return this.Unmount(root, out errorMessage); },
                 "Unmounting"))
-            {
-                this.ReportErrorAndExit(errorMessage);
-            }
+                {
+                    this.ReportErrorAndExit(errorMessage);
+                }
 
-            if (!this.Unattended && 
-                !this.SkipUnregister &&
-                GVFSPlatform.Instance.SupportsGVFSService)
-            {
-                if (!this.ShowStatusWhileRunning(
-                    () => { return this.UnregisterRepo(root, out errorMessage); },
-                    "Unregistering automount"))
+                if (!this.Unattended &&
+                    !this.SkipUnregister &&
+                    GVFSPlatform.Instance.SupportsGVFSService)
                 {
-                    this.Output.WriteLine("    WARNING: " + errorMessage);
+                    if (!this.ShowStatusWhileRunning(
+                        () => { return this.UnregisterRepo(root, out errorMessage); },
+                        "Unregistering automount"))
+                    {
+                        this.Output.WriteLine("    WARNING: " + errorMessage);
+                    }

```

Comment:
```
If we have a Mount that isn't marked as either Successful / Unsuccessful above, it could be because
...
   * It was force killed / BSOD / Power Outage / etc..
      How much effort do we want to put into this scenarios? My preliminary thoughts are to see 
      how common it is before digging further
```

This is a nightly occurrence for most of our users in Windows, because they often take nightly updates that require reboots. So their mount processes are almost always killed as a result of a reboot, not from a clean unmount.

I think we should define a failed mount as a mount that had an error event, _not_ as a mount that is missing an unmount event.

Reply:
@sanoursa interesting!  

So if a machine is turned off it will not have an ending Duration mount event.  So maybe we could make success a bit more loose by just removing the Duration  clause in the query above.  So mount starts without error entries are considered success and starts with error are failure.  Thoughts?

Reply:
Yes that makes sense. Though I was thinking about this, and we may want to think about two classes of failures here. There are mount attempts that just fail outright, e.g. because the driver isn't cooperating, because auth is failing, etc. And then there are mount attempts that succeed, and then later hit some critical error and shut down. We should make sure we are set up to measure each of those independently.

Reply:
Whether or not `"Virtual repo is ready"` was logged by the mount process might help differentiate between these two scenarios.

Reply:
Ahh, I thought we were logging that message to telemetry.  We used to automatically include all events that were `Error` or more important (and this one is `Critical`), but I don't see that code anymore ☹️ .

It looks like it's now `RelatedError` that makes error messages show up in telemetry.

Reply:
@wilbaker any chance you saw the PR that changed this.  Just curious on the history of changing what we move to telemetry